src/api/resource.ts

```diff
@@ -0,0 +1,86 @@
+import Axios from 'axios';
+import { Dispatch } from 'redux';
+
+import {
+  createResourceFailure,
+  createResourceSuccess,
+  getResourceFailure,
+  getResourceSuccess,
+  updateResourceFailure,
+  updateResourceSuccess,
+} from '../redux/actions/resourceActions';
+import { Resource, ResourceFileMap, ResourceMetadata } from '../types/resource.types';
+import { API_BASE_URL, StatusCode } from '../utils/constants';
+
+export const createNewResource = (files: ResourceFileMap[], metadata: ResourceMetadata, owner: string) => {
+  const resource: Partial<Resource> = {
+    createdDate: new Date().toISOString(),
+    files,
+    metadata,
+    owner,
+  };
+
+  return async (dispatch: Dispatch) => {
+    Axios({
+      method: 'POST',
+      url: `/${API_BASE_URL.RESOURCES}`,
+      data: resource,
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'aplication/json',
+      },
+    })
+      .then(response => {
```

Comment:
Har vi noe standpunkt for async/await vs then? Personlig liker jeg hvertfall async/await bedre

Reply:
vi har ikke bestemt om vi skal bruke enten eller. leste en interessant post her: https://itnext.io/javascripts-async-await-versus-promise-the-great-debate-6308cb2e10b3 som sier at async/await er kjappere, så kanskje vi skal bruke det fra nå av?

Reply:
da tar vi async/await